tooling/templatize/internal/end2end/e2e.go

```diff
@@ -170,9 +170,11 @@ func (e *e2eImpl) Persist() (*run.RawRunOptions, error) {
 				return nil, err
 			}
 
-			err = os.WriteFile(e.tmpdir+"/"+b.paramFileName, []byte(b.paramFile), 0644)
-			if err != nil {
-				return nil, err
+			if (len(b.paramFile) != 0) || (len(b.paramFileName) != 0) {
```

Comment:
I'm not sure I follow this logic - in what case is either the file name or the content empty? If the content is empty, why would we care? If the file name is empty but the content is non-empty, why should we write an empty file at `e.tempDir+"/"`?

Reply:
This is for the modules that don't have parameter files. I can change this to fail here if the filename is empty and the content is not, but I figured we would already return an error on line 176 in that case